cgsmiles/resolve.py

```diff
@@ -94,58 +94,67 @@ def __init__(self,
             the cgsmiles string to resolve
         meta_graph: `:class:nx.Graph`
             a potential higher resolution graph
-        fragment_dict: dict[`:class:nx.Garph`]
-            a dict of fragment graphs
+        fragment_dicts: list[dict[str, nx.Graph]]
+            a dict of fragment graphs per resolution
         last_all_atom: bool
             if the last resolution is at the all
-            atom level. Default: True
+            atom level. If True the code will use
+            pysmiles to parse the fragments and
+            return the all-atom molecule.
+            Default: True
         """
         # this is the dict with the fragments
         # either provided and/or extracted
         # from the fragment string
-        self.fragment_dict = fragment_dict
-        # this is the current resolution graph
-        self.molecule = nx.Graph()
+        self.fragment_dicts = fragment_dicts
         # this is the current meta_graph
         self.meta_graph = nx.Graph()
-        # these are possible fragment strings
-        self.fragment_strs = []
 
         # here we figure out how many resolutions
         # we are dealing with
         elements = re.findall(r"\{[^\}]+\}", pattern)
 
-        # case 1)
-        # a meta_graph is provided which means we only
-        # have fragments to deal with
-        if meta_graph:
-            self.fragment_strs = elements
-        # case 2) we have a meta graph only described
+        # case 1) we have a meta graph only described
         # and the fragment come from elsewhere
-        elif len(elements) == 1 and self.fragment_dict:
+        if len(elements) == 1 and self.fragment_dicts:
             self.molecule = read_cgsmiles(elements[0])
-        # case 3) a string containing both fragments and
-        # the meta sequence is provided
+            self.fragment_strs = None
+            # the number of resolutions available
+            self.resolutions = len(self.fragment_dicts)
+            # turn the framgent strings into an iterator
+            self.fragment_strs = None
+            self.fragment_dicts = iter(self.fragment_dicts)
```

Comment:
```suggestion
            # the number of resolutions available
            self.resolutions = len(self.fragment_dicts)
            self.fragment_strs = []
            self.molecule = nx.Graph()
            self.fragment_dicts = self.fragment_dicts
```

Reply:
This branch does not define a self.molecule

Reply:
yes it does in line 120, 133, 137.

Reply:
or you mean when no case is triggerd? That should be impossible though or better raise an error

Reply:
No you're right. I missed the case on 120